packages/server/src/subscriptions/websockets.ts

```diff
@@ -22,12 +22,86 @@ export interface BindWithTokenMsg extends BaseSubscriptionClientMsg {
   payload: { token: string };
 }
 
+const wsToSubLookup = new Map<ws.WebSocket, Set<string>>();
+const subToWsLookup = new Map<string, Set<ws.WebSocket>>();
+let redisSubscriber: Redis | undefined;
+let heartbeatHandler: (() => void) | undefined;
+
+async function setupSubscriptionHandler(): Promise<void> {
+  redisSubscriber = getRedisSubscriber();
+  redisSubscriber.on('message', (channel: string, events: string) => {
+    globalLogger.debug('[WS] redis subscription events', { channel, events });
+    const subEventArgsArr = JSON.parse(events) as [Resource, subscriptionId: string, options?: SubEventsOptions][];
+    for (const [resource, subscriptionId, options] of subEventArgsArr) {
+      const bundle = createSubEventNotification(resource, subscriptionId, options);
+      for (const socket of subToWsLookup.get(subscriptionId) ?? []) {
+        socket.send(JSON.stringify(bundle), { binary: false });
+      }
+    }
+  });
+  await redisSubscriber.subscribe('medplum:subscriptions:r4:websockets');
+}
+
+function subscribeWsToSubscription(ws: ws.WebSocket, subscriptionId: string): void {
+  let wsSet = subToWsLookup.get(subscriptionId);
```

Comment:
🤔  Is it allowed to use the same `Subscription` on multiple websocket connections? 🤔

Reply:
It's technically allowed by the spec, and  Currently our client doesn't support it but it shouldn't be hard to add support for it in the future. I kinda left it open more for other client implementers / to be more FHIR spec-compliant